neqo-http3/src/connection_client.rs

```diff
@@ -2364,4 +2430,676 @@ mod tests {
+                    assert!(
+                        (stream_id == 2)
+                            || (stream_id == 6)
+                            || (stream_id == 10)
+                            || (stream_id == 0)
+                    );
+                    if stream_id == 0 {
+                        assert_eq!(stream_type, StreamType::BiDi);
+                    } else {
+                        assert_eq!(stream_type, StreamType::UniDi);
+                    }
+                }
+                ConnectionEvent::RecvStreamReadable { stream_id } => {
+                    if stream_id == 2 {
+                        // the control stream
+                        // Only stream type will be received no Settings frame will be received.
+                        let mut buf = [0u8; 100];
+                        let (amount, fin) = server.stream_recv(stream_id, &mut buf).unwrap();
+                        assert_eq!(fin, false);
+                        assert_eq!(amount, CONTROL_STREAM_DATA.len());
+                        assert_eq!(&buf[..9], CONTROL_STREAM_DATA);
+                        control_stream = true;
+                    } else if stream_id == 6 {
+                        let mut buf = [0u8; 100];
+                        let (amount, fin) = server.stream_recv(stream_id, &mut buf).unwrap();
+                        assert_eq!(fin, false);
+                        // the remote encoder stream may send a change capacity instruction
+                        // (0x3f, 0x45 = change capacity to 100)
+                        if expect_cap_instruction {
+                            // with a instruction
+                            assert_eq!(amount, 3);
+                            assert_eq!(buf[..3], [0x02, 0x3f, 0x45]);
+                        } else {
+                            // without a instruction
+                            assert_eq!(amount, 1);
+                            assert_eq!(buf[..1], [0x02]);
+                        }
+                        qpack_encoder_stream = true;
+                    } else if stream_id == 10 {
+                        let mut buf = [0u8; 100];
+                        let (amount, fin) = server.stream_recv(stream_id, &mut buf).unwrap();
+                        assert_eq!(fin, false);
+                        assert_eq!(amount, 1);
+                        assert_eq!(buf[..1], [0x3]);
+                        qpack_decoder_stream = true;
+                    } else if stream_id == 0 {
+                        assert!(expect_request);
+                        check_header_frame(&mut server, stream_id, true);
+                        let _ = server.stream_send(
+                            stream_id,
+                            &[
+                                // headers
+                                0x01, 0x06, 0x00, 0x00, 0xd9, 0x54, 0x01, 0x33,
+                                // the data frame
+                                0x0, 0x3, 0x61, 0x62, 0x63,
```

Comment:
As before, if you are using a standard request, then you should define the shape of the request and its expected serialization together so that people who have to maintain these tests have an easier time if they want to change something.